<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1507</width>
    <height>820</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QPushButton" name="openImage">
        <property name="text">
         <string>Open Image</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="openCamera">
        <property name="text">
         <string>Start Camera</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="takePhoto">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Take Photo</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="usesScrollButtons">
       <bool>false</bool>
      </property>
      <property name="tabsClosable">
       <bool>false</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="Image">
       <attribute name="title">
        <string>Image</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QGraphicsView" name="imageGraphicsView">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QFormLayout" name="imageSettingsFormLayout">
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::ExpandingFieldsGrow</enum>
          </property>
          <property name="labelAlignment">
           <set>Qt::AlignJustify|Qt::AlignVCenter</set>
          </property>
          <property name="formAlignment">
           <set>Qt::AlignJustify|Qt::AlignVCenter</set>
          </property>
          <item row="0" column="0">
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <property name="rightMargin">
             <number>12</number>
            </property>
            <item>
             <widget class="QCheckBox" name="cartoonifyCheckBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Cartoonify</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
              <property name="tristate">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="sharpenCheckbox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Sharpen</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="blurCheckBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Blur</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="gaussianBlurCheckBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Gaussian Blur</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="0" column="1">
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <property name="leftMargin">
             <number>12</number>
            </property>
            <item>
             <widget class="QSlider" name="cartoonifyHS">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>99</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksBelow</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="sharpenHS">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksBelow</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="blurHS">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>99</number>
              </property>
              <property name="singleStep">
               <number>2</number>
              </property>
              <property name="sliderPosition">
               <number>1</number>
              </property>
              <property name="tracking">
               <bool>true</bool>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="invertedAppearance">
               <bool>false</bool>
              </property>
              <property name="invertedControls">
               <bool>false</bool>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksBelow</enum>
              </property>
              <property name="tickInterval">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="gaussianBlurHS">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::TicksBelow</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="Video">
       <attribute name="title">
        <string>Video</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_12">
        <item>
         <widget class="QGraphicsView" name="videoGraphicsView"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QCheckBox" name="addMouse">
            <property name="text">
             <string>Mouse</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="addMoustache">
            <property name="text">
             <string>Moustache</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="addSpectacles">
            <property name="text">
             <string>Spectacles</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="addElephantFace">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>Elephant Face</string>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="Text">
       <attribute name="title">
        <string>Text</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QSplitter" name="splitter">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <widget class="QGraphicsView" name="textEditorImageGraphicsView"/>
          <widget class="QTextEdit" name="textAreaForOCR"/>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="extractText">
          <property name="text">
           <string>Extract Text</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="OD">
       <attribute name="title">
        <string>Object Detection</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_11">
        <item>
         <widget class="QGraphicsView" name="graphicsViewOD"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="enabled">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1507</width>
     <height>37</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuImage">
    <property name="title">
     <string>Image</string>
    </property>
    <addaction name="actionOpen_Image"/>
    <addaction name="actionSave_Image"/>
    <addaction name="actionSave_Image_As"/>
    <addaction name="actionZoom_In"/>
    <addaction name="actionZoom_Out"/>
    <addaction name="actionFlip_Horizontal"/>
    <addaction name="actionFlip_Vertical"/>
    <addaction name="actionRotate_Left"/>
    <addaction name="actionRotate_Right"/>
    <addaction name="actionFit_to_Window"/>
    <addaction name="actionNormal_Size"/>
    <addaction name="actionClose_Image"/>
   </widget>
   <widget class="QMenu" name="menuVideo">
    <property name="title">
     <string>Video</string>
    </property>
    <widget class="QMenu" name="menuFilters">
     <property name="title">
      <string>Filters</string>
     </property>
     <addaction name="actionCartoonify"/>
     <addaction name="actionBlur"/>
     <addaction name="actionSharpen"/>
    </widget>
    <widget class="QMenu" name="menuAdd_Face_Masks">
     <property name="title">
      <string>Add Face Masks</string>
     </property>
     <addaction name="actionMoustache"/>
     <addaction name="actionSpectacles"/>
     <addaction name="actionMonkey"/>
     <addaction name="actionMouse_Nose"/>
     <addaction name="actionDuck_Face"/>
    </widget>
    <addaction name="actionOpen_Camera"/>
    <addaction name="actionClose_Camera"/>
    <addaction name="menuFilters"/>
    <addaction name="menuAdd_Face_Masks"/>
   </widget>
   <widget class="QMenu" name="menuText">
    <property name="title">
     <string>Text</string>
    </property>
    <addaction name="actionSave_Text"/>
    <addaction name="actionOCR"/>
   </widget>
   <widget class="QMenu" name="menuObject_Detection">
    <property name="title">
     <string>Object Detection</string>
    </property>
    <widget class="QMenu" name="menuYolo">
     <property name="title">
      <string>Yolo</string>
     </property>
     <addaction name="actionCar_Detection_2"/>
     <addaction name="actionPet_Detection_2"/>
    </widget>
    <addaction name="menuYolo"/>
    <addaction name="actionSSD"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionShow_Cameras"/>
   </widget>
   <addaction name="menuImage"/>
   <addaction name="menuVideo"/>
   <addaction name="menuText"/>
   <addaction name="menuObject_Detection"/>
   <addaction name="menuAbout"/>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionOpen_Camera">
   <property name="text">
    <string>Open Camera</string>
   </property>
  </action>
  <action name="actionClose_Camera">
   <property name="text">
    <string>Close Camera</string>
   </property>
  </action>
  <action name="actionCartoonify">
   <property name="text">
    <string>Cartoonify</string>
   </property>
  </action>
  <action name="actionAdd_face_Masks">
   <property name="text">
    <string>Add face Masks</string>
   </property>
  </action>
  <action name="actionBlur">
   <property name="text">
    <string>Blur</string>
   </property>
  </action>
  <action name="actionSharpen">
   <property name="text">
    <string>Sharpen</string>
   </property>
  </action>
  <action name="actionMoustache">
   <property name="text">
    <string>Moustache</string>
   </property>
  </action>
  <action name="actionSpectacles">
   <property name="text">
    <string>Spectacles</string>
   </property>
  </action>
  <action name="actionMonkey">
   <property name="text">
    <string>Elephant Face</string>
   </property>
  </action>
  <action name="actionMouse_Nose">
   <property name="text">
    <string>Mouse Nose</string>
   </property>
  </action>
  <action name="actionDuck_Face">
   <property name="text">
    <string>Duck Face</string>
   </property>
  </action>
  <action name="actionSharpen_2">
   <property name="text">
    <string>Sharpen</string>
   </property>
  </action>
  <action name="actionDilate">
   <property name="text">
    <string>Dilate</string>
   </property>
  </action>
  <action name="actionBlur_2">
   <property name="text">
    <string>Blur</string>
   </property>
  </action>
  <action name="actionGaussian_Blur">
   <property name="text">
    <string>Gaussian Blur</string>
   </property>
  </action>
  <action name="actionCar_Detection">
   <property name="text">
    <string>Car Detection</string>
   </property>
  </action>
  <action name="actionPet_Detection">
   <property name="text">
    <string>Pet Detection</string>
   </property>
  </action>
  <action name="actionFlower_Classifier">
   <property name="text">
    <string>Flower Classifier</string>
   </property>
  </action>
  <action name="actionOCR">
   <property name="text">
    <string>OCR Settiings</string>
   </property>
  </action>
  <action name="actionImage_Text_Detection">
   <property name="text">
    <string>Image Text Detection</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="text">
    <string>Undo</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As</string>
   </property>
  </action>
  <action name="actionCar_Detection_2">
   <property name="text">
    <string>Car Detection</string>
   </property>
  </action>
  <action name="actionPet_Detection_2">
   <property name="text">
    <string>Pet Detection</string>
   </property>
  </action>
  <action name="actionSSD">
   <property name="text">
    <string>SSD</string>
   </property>
  </action>
  <action name="actionOpen_Image">
   <property name="text">
    <string>Open Image</string>
   </property>
  </action>
  <action name="actionSave_Image">
   <property name="text">
    <string>Save Image</string>
   </property>
  </action>
  <action name="actionSave_Image_As">
   <property name="text">
    <string>Save Image As</string>
   </property>
  </action>
  <action name="actionClose_Image">
   <property name="text">
    <string>Close Image</string>
   </property>
  </action>
  <action name="actionSave_Text">
   <property name="text">
    <string>Save Text</string>
   </property>
  </action>
  <action name="actionChoose_Model">
   <property name="text">
    <string>Choose Model</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionShow_Cameras">
   <property name="text">
    <string>Show Cameras</string>
   </property>
  </action>
  <action name="actionZoom_In">
   <property name="text">
    <string>Zoom In</string>
   </property>
  </action>
  <action name="actionZoom_Out">
   <property name="text">
    <string>Zoom Out</string>
   </property>
  </action>
  <action name="actionFlip_Horizontal">
   <property name="text">
    <string>Flip Horizontal</string>
   </property>
  </action>
  <action name="actionFlip_Vertical">
   <property name="text">
    <string>Flip Vertical</string>
   </property>
  </action>
  <action name="actionRotate_Left">
   <property name="text">
    <string>Rotate Left</string>
   </property>
  </action>
  <action name="actionRotate_Right">
   <property name="text">
    <string>Rotate Right</string>
   </property>
  </action>
  <action name="actionFit_to_Window">
   <property name="text">
    <string>Fit to Window</string>
   </property>
  </action>
  <action name="actionNormal_Size">
   <property name="text">
    <string>Normal Size</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
